File: Week4/zad4.php
<?php
class Player {
    public $name;
    public $type;
    public $movementSpeed;
    public $damage;

    function __construct($name,$type){
        $this->name = $name;
        $this->type = $type;
    }
    function set_movSpeed($speed){
        $this->movementSpeed = $speed;
    }
    function set_damage($damage){
        $this->damage = $damage;
    }
    function printplayer(){
        echo "Name: ".$this->name . ",Type: ". $this->type . ", Speed: " .$this->movementSpeed . ", Damage: ".$this->damage . ".\n";
    }
}
function spawnPlayer($name,$type){
    $allTypes = array("human"=>[100,120,90,100], "ork"=>[90,100,45,60],"elf"=>[110,130,35,45],"undead"=>[90,120,40,50]);
    $player = new Player($name,$type);

    $player->set_movSpeed(moveSpeed($type,$allTypes));
    $player->set_damage(setDamage($type,$allTypes));
    $player->printplayer();

}
function moveSpeed($type,$all){
    return rand($all[$type][0],$all[$type][1]);
}
function setDamage($type,$all){
    return rand($all[$type][2],$all[$type][3]);
}
spawnPlayer("stefanj","ork")


?>

File: Python/zad4.py
import random

allTypes = {
    "human": {"minSpeed": 100, "maxSpeed": 120, "minDamage": 40, "maxDamage": 45},
    "ork": {"minSpeed": 90, "maxSpeed": 100, "minDamage": 45, "maxDamage": 60},
    "elf": {"minSpeed": 110, "maxSpeed": 130, "minDamage": 35, "maxDamage": 45},
    "undead": {"minSpeed": 90, "maxSpeed": 120, "minDamage": 40, "maxDamage": 50},
}


def spawnPlayer(name, type):
    obj = player(name, type)
    return obj


class player:
    def __init__(self, name, type):
        self.name = name
        self.type = type
        self.movementSpeed = movSpeed(type, allTypes)
        self.damage = typeDamage(type, allTypes)

    def __str__(self):
        return f"Name: {self.name}, Type: {self.type}, Movement speed: {self.movementSpeed}, Damage: {self.damage}."


def movSpeed(type,allTypes):
    return random.randrange(allTypes[type]["minSpeed"], allTypes[type]["maxSpeed"])


def typeDamage(type,allTypes):
    return random.randrange(allTypes[type]["minDamage"], allTypes[type]["maxDamage"])


print(spawnPlayer("stefanj", "undead"))
